samples/sample_common/src/plugin_utils.cpp

```diff
@@ -120,7 +120,7 @@ const mfxPluginUID & msdkGetPluginUID(mfxIMPL impl, msdkComponentType type, mfxU
             break;
         }
     }
-    else if (impl |= MFX_IMPL_HARDWARE)
+    else if ((impl |= MFX_IMPL_HARDWARE))
```

Comment:
Really looks invalid. Always true if MFX_IMPL_HARDWARE !=0. Also modification of impl is useless here.